File: Cryptography/monoAlphabeticCipher.go
package main

import "fmt"

func mac(text string) string {
	subTable := map[rune]rune{
		'a': 'q',
		'b': 'w',
		'c': 'x',
		'd': 'r',
		'e': 't',
		'f': 'y',
		'g': 'u',
		'h': 's',
		'i': 'o',
		'j': 'p',
		'k': 'n',
		'l': 'v',
		'm': 'z',
	}
	result := ""
	for _, letter := range text {
		value, found := subTable[letter]
		if found {
			result += string(value)
		} else {
			for k, v := range subTable {
				if v == letter {
					result += string(k)

				}
			}
		}

	}

	return result

}

func task3() {
	var input string
	fmt.Scanf("%s", &input)
	encrypted := mac(input)
	decrypted := mac(encrypted)
	print("Encrypted: " + encrypted + "\n")
	print("Decrypted: " + decrypted + "\n")
}

File: Cryptography/wasif-sir/monoAlphabeticCipher.cpp
#include <bits/stdc++.h>
using namespace std;
string mac(string text) {
    unordered_map<char, char> subTable = {
        {'a', 'q'},
        {'b', 'w'},
        {'c', 'x'},
        {'d', 'r'},
        {'e', 't'},
        {'f', 'y'},
        {'g', 'u'},
        {'h', 's'},
        {'i', 'o'},
        {'j', 'p'},
        {'k', 'n'},
        {'l', 'v'},
        {'m', 'z'}
    };
    string result = "";
    for (char letter : text) {
        auto found = subTable.find(letter);
        if (found != subTable.end()) {
            result += found->second;
        } else {      
            for (auto& pair : subTable) {
                if (pair.second == letter) {
                    result += pair.first;
                    break;
                }
            }
        }
    }
    return result;
}
int main() {
    string input;
    cout << "Enter Text to be encrypted:" << endl;
    cin >> input;
    string encrypted = mac(input);
    string decrypted = mac(encrypted);
    cout << "Encrypted: " << encrypted << endl;
    cout << "Decrypted: " << decrypted << endl;
    return 0;
}